### ISSUE TITLE
Disbursements and expenditures

### ISSUE THREAD

**Issue by andylolz** [pwyf/2018-technical-consultation#13]
## Description

Individual actual financial disbursements and expenditures must be related to individual activities and must be on a per-transaction basis.

This is only expected if the activity is in the implementation, completion or post-completion phase.

### IATI 2.0x

```python
For each current activity,
i​f `activity-status/@code` i​s one of (2, 3, 4)
then
`transaction/transaction-type[@code="3"]`
should be present
o​r
`transaction/transaction-type[@code="4"]`
should be present
```

### IATI 1.0x

```python
For each current activity,
i​f `activity-status/@code` i​s one of (2, 3, 4)
then
`transaction/transaction-type[@code="D"]`
should be present
o​r
`transaction/transaction-type[@code="E"]`
should be present
```

**Comment by breidertmt** [comment]
Can you confirm there is no penalty if you have an obligation/commitment, but in the same year no disbursement/expenditure. We have multiple instances where something might have been obligated but not disbursed. When we ran data through the tracker, it seemed to be penalizing us for this scenario.

**Comment by andylolz** [comment]
@breidertmt:

> Can you confirm there is no penalty if you have an obligation/commitment, but in the same year no disbursement/expenditure

So, the tests for obligation/commitment and disbursement/expenditure are not tied – there’s no dependence between them.

> We have multiple instances where something might have been obligated but not disbursed.

Is this commonplace? If so, are you able to provide an example of this?

**Comment by breidertmt** [comment]
@andylolz Yes, this is quite common. This is because sometimes awards are across multiple years, say a 5-year award. An obligation might be made in third quarter and disbursed in first quarter of the next year. This is not uncommon.

**Comment by breidertmt** [comment]
![tracker](https://cloud.githubusercontent.com/assets/26721209/24674446/6c637cc2-1949-11e7-9cc4-c0e2fc6adccb.gif)

@andylolz When we ran through the tool we got the result in the screenshot attached. From our read it looked like we were being penalized because a 3 or 4 wasn't present. Can you take a look.

**Comment by andylolz** [comment]
Hi @breidertmt. That’s correct – so, the test expects that all current activities in the implementation, completion or post-completion phase will have either disbursements or expenditures.

Looking up the activities in that screenshot on d-portal, it looks like those activities are all in completion, but don’t have disbursements or expenditures. So as it stands, those activities would not pass this test.

 * [US-11-AF-10-SAQMMA15C0003](http://d-portal.org/q.html?aid=US-11-AF-10-SAQMMA15C0003)
 * [US-11-AF-10-SAQMMA15F0462](http://d-portal.org/q.html?aid=US-11-AF-10-SAQMMA15F0462)
 * [US-11-AF-10-SAQMMA16C0061](http://d-portal.org/q.html?aid=US-11-AF-10-SAQMMA16C0061)
 * [US-11-AF-10-SINLEC15L0121](http://d-portal.org/q.html?aid=US-11-AF-10-SINLEC15L0121)

**Comment by breidertmt** [comment]
@andylolz To clarify the test says if current, check commitment and check for disbusement/expenditure? If commitment, but no disbursement/expenditure then fail. The example above perhaps was a bad one. 

What is legitimate that we see is award in implementation and perhaps an obligation/commitment in the current year, but it is possible that a disbursement/expenditure does not happen in the same year. This is because awards are multipl year. So  not excusing the ones that just don't have a disbursement at all through lifetime, but those who do not have a disbursement in the same FY or current year as the commiment. Hopefully this makes sense.

**Comment by agnessurry** [comment]
Hi, in the case of MDBs, for sovereign operations that are being implemented, disbursement may happen the following year or much later after commitment.

Disbursement cannot be  disclosed in the case of non-sovereign operations.

**Comment by andylolz** [comment]
Hi @breidertmt,

> To clarify the test says if current, check commitment and check for disbusement/expenditure? If commitment, but no disbursement/expenditure then fail.

No, that’s not correct. This test ignores commitments. It tests that all current activities in the implementation, completion or post-completion phase have at least one disbursement or expenditure.

Apologies for the confusion about this.

**Comment by AndieVaughn** [comment]
We realized that to pass this test, nothing must be blank and we must have a 0 in the field.  When obligations or disbursements fall outside of the test year, if there is no 0 in the file, then there is a penalty.  We recommend combining Disbursement and Commitment to measure if there is a commitment OR a disbursement into 1 test so that this issue is resolved.

**Comment by andylolz** [comment]
Thanks for the comments here.

We’ll modify this test to check that at least one non-zero value disbursement or expenditure is present.
